index.js

```diff
@@ -1,121 +1,74 @@
-'use strict';
 const request = require('request-promise-native');
 
 const role_to_use = process.env.ROLE;
 const couch_url = process.env.COUCH_URL;
-let compiledUrl = {};
 
-if(role_to_use === undefined){
+if(role_to_use === undefined) {
     console.log('ROLE is not set, please set it and try again');
     process.exit(1);
 }
-if(couch_url === undefined){
+if(couch_url === undefined) {
     console.log('COUCH_URL is not set, please set it and try again');
     process.exit(1);
 }
 
-const handleError = function(e){
-    if (e.statusCode === 401) {
-        console.log('Bad authentication for CouchDB. Check that COUCH_URL has correct usernames and passwords.');
-        process.exit(1);
-    } else {
-        console.log('Error connecting to CouchDB: ' + e.message);
-        process.exit(1);
-    }
-}
-
-const getAllUsers = function (){
+const compileUrl = (path) => {
     try {
-        compiledUrl = new URL('/_users/_all_docs', couch_url);
+        return new URL(path, couch_url).href;
     } catch(e) {
-        console.log('Error while creating url: ', e.message);
-        process.exit(1);
+        throw new Error(`Error while creating url: ${e.message}`);
     }
+};
 
-    const options = {
-        uri: compiledUrl.href,
-        json: true
-    };
-
-    return request.get(options)
-        .then(users => {
-            if (typeof users.rows === 'object' && users.rows.length > 0){
-                // first user is always admin user, let's not mess with them
-                users.rows.shift();
-                console.log("Found",users.rows.length,"total users.");
-                return users.rows;
-            } else {
-                console.log('No users found');
-                process.exit(1);
-            }
-        })
-        .catch(e => {
-            handleError(e);
-        })
+const getAllUsers = async () => {
+    const uri = compileUrl('/_users/_all_docs');
+    return request.get({ uri, json: true});
 }
 
-const getUserPerms = function (allUsers){
-    allUsers.forEach((username)=>{
```

Comment:
Without an await (or other control) these requests were happening in parallel. It would be easy to DoS the server this way, so I've made them run serially. Even better would be to use the bulk api to update a whole lot of users in one request.

Reply:
interesting! I tried not to do this as async > sync...right!?!? ;)

That said, I did try this on an instance with 1000+ users and it didn't tip over.  Might work differently in production!